devito/types/sparse.py

```diff
@@ -35,14 +35,13 @@ class AbstractSparseFunction(DiscreteFunction, Differentiable):
     _sub_functions = ()
     """SubFunctions encapsulated within this AbstractSparseFunction."""
 
-    def __init__(self, *args, **kwargs):
-        if not self._cached():
-            super(AbstractSparseFunction, self).__init__(*args, **kwargs)
-            self._npoint = kwargs['npoint']
-            self._space_order = kwargs.get('space_order', 0)
+    def __init_finalize__(self, *args, **kwargs):
```

Comment:
note for reviewers: this is exactly like before, but de-indented as we don't need ``if not self._cached()`` anymore